lib/schedule.js

```diff
@@ -186,9 +186,31 @@ Job.prototype.schedule = function(spec) {
   var self = this;
   var success = false;
   var inv;
+  var start;
+  var end;
+
+  if (typeof spec === 'object' && spec.rule) {
+    start = spec.start || null;
+    end = spec.end || null;
+    spec = spec.rule;
+
+    if (start != null) {
+      if (!(start instanceof Date)) {
+        start = new Date(start);
+      }
+      if (!isValidDate(start) || start.getTime() < Date.now()) {
+        start = null;
+      }
+    }
+
+    if (end != null && !(end instanceof Date) && !isValidDate(end = new Date(end))) {
+      end = null;
+    }
+  }
+
   try {
-    var res = cronParser.parseExpression(spec);
-    inv = scheduleNextRecurrence(res, self);
+    var res = cronParser.parseExpression(spec, { currentDate: start });
```

Comment:
when using the endDate cronParser,it will  throw an error as the endDate is less than the startDate.
And execute function scheduleNextRecurrence later, so i think that not using it.
we can handle all the logic of the endDate in function scheduleNextRecurrence.

If you can read chinese, this is a wonderful thing for me，/(ㄒoㄒ)/~~